kernel/src/hil/hmac.rs

```diff
@@ -0,0 +1,39 @@
+//! Interface for HMAC
+
+use crate::returncode::ReturnCode;
+
+/// Implement this trait and use `set_client()` in order to receive callbacks.
+pub trait Client<'a> {
+    // On success the `result` will contain the digest. On error it will contain
+    // the HMAC error value.
+    fn hash_done(&'a self, result: Result<&'a mut [u8], u32>);
+}
+
+pub const HMAC_KEY_SIZE: usize = 16;
+
+pub trait HMAC<'a> {
+    /// Enable the HMAC hardware.
+    /// Must be called before any other methods
+    fn init(&self);
```

Comment:
I'm also working on using the ARM CryptoCell-310 HMAC accelerator and wanted to make sure the HIL was neither duplicating nor incompatible with the OT one.
In your HIL, how do we set the underlying hash algorithm? If the HIL is only compatible with 1 algorithm I suggest to then rename it HmacSha512 (according to the `HMAC_KEY_SIZE` value)

Reply:
OT only supports SHA256. I can add something to allow specifying the hash algorithm.